composer.json

```diff
@@ -20,20 +20,22 @@
             "homepage": "https://github.com/friendsofsymfony/FOSHttpCache/contributors"
         }
     ],
+    "minimum-stability": "dev",
+    "prefer-stable": true,
     "require": {
-        "php": ">=5.4.8",
+        "php": ">=5.5.9",
         "symfony/event-dispatcher": "^2.3||^3.0",
         "symfony/options-resolver": "^2.3||^3.0",
-        "psr/http-message-implementation": "~1.0",
-        "php-http/adapter-implementation": "^0.1.0",
-        "php-http/discovery": "^0.1.1",
-        "php-http/message-decorator": "^0.1.0"
+        "php-http/client-implementation": "^1.0.0",
+        "php-http/discovery": "^0.6.0",
+        "php-http/message": "^0.2.1",
+        "puli/cli": "^1.0.0-beta8"
```

Comment:
@dbu Yes, Discovery should be optional with FOSHttpCache. We can make it so in two ways:

1. just move it from `require` to `suggest` and document it properly; and maybe add conditional checks around the Discovery calls in our code
2. remove all Discovery calls from our code and document that users should either pass an HTTP client explicitly or do the discovery in their own code.

Against 2 and @sagikazarmark’s idea to have users install Puli is that it becomes a lot more work for users to get started with FOSHttpCache. 

For discovery’s Puli dependency, see https://github.com/php-http/discovery/issues/36.